src/common/mailing.ts

```diff
@@ -115,69 +115,75 @@ export const syncSubscription = async function (
     { ids: userIds },
   );
 
+  if (!Array.isArray(userAttributes?.customers)) {
+    return;
+  }
+
+  if (userAttributes.customers.length === 0) {
+    return;
+  }
+
   await con.transaction(async (manager) => {
-    userAttributes?.customers.forEach(
-      async (customer: {
-        id: string;
-        attributes: { cio_subscription_preferences: string };
-        unsubscribed: boolean;
-      }) => {
-        const subs = JSON.parse(
-          customer?.attributes?.cio_subscription_preferences || '{}',
-        );
-        const unsubscribed = customer?.unsubscribed;
-        const digest =
-          isSubscribed(subs, CioUnsubscribeTopic.Digest) && !unsubscribed;
+    for (const customer of userAttributes.customers as {
```

Comment:
just replace forEach with for of, latter actually waits for async callback which fixes the issue